main.c

```diff
@@ -0,0 +1,262 @@
+
+	GPIOE->MODER |= (GPIO_MODER_MODE8_1); //Setting the GPIOE mode register to alternate function mode for pin 8
+
+	GPIOE->AFR[1] &= ~(0xF); //Selecting AF mode 1
+
+	GPIOE->AFR[1] |= 0x1; //Selecting AF mode 1
+
+	GPIOE->OSPEEDER &= ~(GPIO_OSPEEDER_OSPEEDR8); //Seeting output speed 
+
+	GPIOE->PUPDR &= ~(GPIO_PUPDR_PUPD8); //Selecting no pull up or pull down
+
+}
+
+void timer1init(){
+
+	RCC->APB2ENR |= RCC_APB2ENR_TIM1EN; // Enable the timer 1 clock
+	TIM1->CR1 &= ~TIM_CR1_DIR; //Select up counting
+	TIM1->PSC = 159; //change this value for whatever we need
+	TIM1->ARR = 999; //Set PWM period change this value
+	TIM1->CCMR1 &= ~TIM_CCMR1_OC1M; //Clear output compare mode bits for channel 1
+	TIM1->CCMR1 |= TIM_CCMR1_OC1M; //Select PWM mode 2 oupt on Channel 1
+	TIM1->CCMR1 |= TIM_CCMR1_OC1PE; //Output 1 preload enable
+	TIM1->CCER &= ~TIM_CCER_CC1NP; //Select output polarity
+	TIM1->CCER |= TIM_CCER_CC1NE; //Enable complemtary output of channel 1 
+	TIM1->BDTR |= TIM_BDTR_MOE; //Main output enable
+	TIM1->CCR1 = 500; //initial duty cycle change this value
+	TIM1->CR1 |= TIM1_CR1_CEN; //Enable TIM1
+
+}
+
+void SysTick_Initialize(uint32_t ticks)
+{
+	SysTick->CTRL = 0; // disable counter and IRQ
+	SysTick->LOAD = ticks - 1; // set the load register
+	//make systick least urgent
+	NVIC_SetPriority(SysTick_IRQn, (1<<__NVIC_PRIO_BITS)-1);
+	SysTick->VAL = 0; // reset counter value
+	SysTick->CTRL |= SysTick_CTRL_CLKSOURCE_Msk; // 1 = processor clock 0=ext clk
+	SysTick->CTRL |= SysTick_CTRL_TICKINT_Msk; // enable systick interrupt
+	SysTick->CTRL |= SysTick_CTRL_ENABLE_Msk; // enable timer
+}
+
+void moveVertical(uint8_t distance, uint8_t direction, uint8_t speed){ //our function that prints the char to the lcd if a button is pressed
+	
+		
+		SysTick_Initialize(motorSpeed / speed); // reload value for fast wind up
+		motorTarget = (distance % 60) * 69; //gets the correct final motor location
+		direction = this->direction; //sets the direction
+		SysTick->CTRL |= SysTick_CTRL_ENABLE_Msk; //enables the systick timer
+	
+	
+}
+
+void moveHorizontal(uint8_t* distance, uint8_t direction, uint8_t* speed){ //our function that prints the char to the lcd if a button is pressed
+	
+		
+		SysTick_Initialize(motorSpeed / this->speed); // reload value for fast wind up
+		motorTarget = (distance % 60) * 69; //gets the correct final motor location
+		direction = this->direction; //sets the direction
+		SysTick->CTRL |= SysTick_CTRL_ENABLE_Msk; //enables the systick timer
+	
+	
+}
+
+void goTo(uint8_t posx, uint8_t posy){
+
+	int tempx = motorLocationX - posx;
+	int tempy = motorLocationY - posy;
+
+	if (tempx < 0){
+		direction = 1;
+		tempx *= -1;
+	}
+	else{
+		direction = -1;
+	}
+	
+	moveHorizontal(tempx, direction, 3);
+	while(motorLocationX != posx);
+
+	if (tempy < 0){
+		direction = 1;
+		tempy *= -1;
+	}
+	else{
+		direction = 1;
+	}
+	
+	moveVertical(tempx, direction, 3);
+	while(motorLocationY != posy);
+
+
+}
+
+void home(void){
```

Comment:
This is probably a good placeholder for now, but the purpose of the home function is to zero the values in the motor location values to a reliable location on the physical x-y plane. Perhaps the actual sensing of the homing switches could be handled in the sysTickHandler or something, but we would need to make sure that the motors make it all the way there.